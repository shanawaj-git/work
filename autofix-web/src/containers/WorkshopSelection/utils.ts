import {
  BoundPayloadType,
  GetDirections,
  LatLngType,
  MapObjType,
} from "./interfaces";

export const markerIcon = (icon: string) => ({
  scaledSize: {
    height: 24,
    width: 24,
  },
  url: icon,
});

export const getBoundLocation = ({
  map,
  onBoundChange,
}: {
  map: MapObjType;
  onBoundChange: (e: BoundPayloadType) => void;
}) => {
  const bounds = map.getBounds();
  const southWest = getSouthWestCoordinates(bounds);
  const northEast = getNorthEastCoordinates(bounds);
  const result = buildBoundPayload(northEast, southWest);
  onBoundChange(result);
};

const getSouthWestCoordinates = (bounds: GetDirections): LatLngType => ({
  lat: bounds.getSouthWest().lat(),
  lng: bounds.getSouthWest().lng(),
});

const getNorthEastCoordinates = (bounds: GetDirections): LatLngType => ({
  lat: bounds.getNorthEast().lat(),
  lng: bounds.getNorthEast().lng(),
});

const buildBoundPayload = (
  northEast: LatLngType,
  southWest: LatLngType
): BoundPayloadType => ({
  locationBoxBounds: {
    bottomLeft: {
      type: "Point",
      coordinates: [southWest.lng, southWest.lat],
    },
    topRight: { type: "Point", coordinates: [northEast.lng, northEast.lat] },
  },
});
