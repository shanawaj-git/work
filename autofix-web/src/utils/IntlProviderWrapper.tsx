import React, { useMemo } from "react";
import { getTranslation, TranslationsType } from "../translations";
import { IntlProvider } from "react-intl";
import { flatData, I18NLocale } from ".";

export const withWrapper = (component: JSX.Element, pageProps = { en: {} }) => (
  <IntlProviderWrapper locale="en" pageProps={pageProps}>
    {component}
  </IntlProviderWrapper>
);

export const IntlProviderWrapper = ({
  children,
  pageProps,
  locale,
}: {
  children: JSX.Element;
  pageProps: TranslationsType;
  locale: I18NLocale;
}) => {
  const messages = useMemo(() => {
    return {
      ...getTranslation(locale),
      ...flatData(pageProps[locale] || {}),
    };
  }, [locale, pageProps]);

  return (
    <IntlProvider locale={`${locale}`} messages={messages}>
      {children}
    </IntlProvider>
  );
};
