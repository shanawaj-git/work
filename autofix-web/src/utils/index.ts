import { itemType } from "@/containers/RecommendedMaintenance";
import {
  AssetType,
  fetchCommonData,
  FooterType,
  getAssets,
  getAssetsData,
  getLabelData,
  getLabels,
  HomeActionType,
  LabelType,
  ServiceType,
} from "@/services/common";
import { TestimonialType } from "@/services/home/testimonial";

export const utils = "UTILS";
const GRAPHQL_CLIENT_READ_TIMEOUT = 5000;

interface getCommonPageContentParams {
  locale: I18NLocale;
  pageName: string;
}

export interface HomeContentType {
  labels?: LabelType[];
  assets?: AssetType[];
  services?: ServiceType[];
  actions?: HomeActionType[];
  testimonials?: TestimonialType[];
  footer?: FooterType;
}

export type I18NLocale = "en" | "ar";

/**
 * Will Return a Data object with Labels and Assets Regarding the page from Strapi
 * @param  locale , pageName
 * @returns Object with Labels and Assets
 */

export const getCommonPageContent = async ({
  locale,
  pageName,
}: getCommonPageContentParams): Promise<HomeContentType> => {
  let data: HomeContentType = {};
  return new Promise(async (resolve, reject) => {
    try {
      const commonData = await (
        await fetchCommonData(locale as I18NLocale, pageName)
      ).data;
      const assets = getAssets(commonData);
      const assetsData = getAssetsData(assets);
      const labels = getLabelData(getLabels(commonData));
      data = { assets: assetsData, labels };
      resolve(data);
    } catch (error) {
      resolve(data);
    }
  });
};

interface rejectEarlyType<T> {
  fn: () => Promise<T>;
  def: T;
  timeout?: number;
}

export const rejectEarly = <T>({
  fn,
  def,
  timeout = GRAPHQL_CLIENT_READ_TIMEOUT,
}: rejectEarlyType<T>): Promise<T> => {
  return new Promise(async (resolve, _reject) => {
    setTimeout(() => {
      resolve(def);
    }, timeout);
    try {
      const out: T = await fn();
      resolve(out);
    } catch (error) {
      resolve(def);
    }
  });
};

export const validateEmail = (email: string) =>
  /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(email);

export const flatData = (obj: { [key: string]: {} }) =>
  Object.values(obj).reduce((acc: {}, curr: {}) => {
    return { ...acc, ...curr };
  }, {});

export const buildImageUrl = (path: string): string =>
  process.env.NEXT_PUBLIC_STRAPI_SERVICE?.concat(path) || "";

export const groupBy = <T>(
  array: T[],
  predicate: (value: T, index: number, array: T[]) => string
) =>
  array.reduce((acc, value, index, array) => {
    (acc[predicate(value, index, array)] ||= []).push(value);
    return acc;
  }, {} as { [key: string]: T[] });

export const includesCaseInsensitive = (
  string: string,
  substring: string
): Boolean => {
  return string.toLowerCase().includes(substring.toLowerCase());
};
export const getTotalForMax = (data: itemType[]) => {
  return data.reduce((acc: number, curr: { cost: string }) => {
    let value = acc;
    value += +parseFloat(
      (curr.cost.match(/[0-9]+/gm) || ["0"]).splice(-1)[0]
    ).toFixed(2);
    return value;
  }, 0);
};

export const getCurrency = (data: itemType[]) => {
  return (data[0].cost.match(/[A-Z]+/gm) || ["USD"]).splice(-1)[0];
};

export const getCurrentLocation = () => {
  return new Promise((resolve: (e: GeolocationCoordinates) => void, reject) => {
    if (!navigator) throw new Error("Navigator doesn't exist");
    if (!navigator.geolocation) throw new Error("Geolocation doesn't exist");
    navigator.geolocation.getCurrentPosition(
      (position: GeolocationPosition) => {
        return resolve(position.coords);
      },
      reject,
      {
        maximumAge: 5000,
        enableHighAccuracy: true,
      }
    );
  });
};

interface getCurrentLiveLocationType {
  callBack: (e: GeolocationPosition) => void;
  onError: (e: GeolocationPositionError | Error) => void;
}

export const getCurrentLiveLocation = ({
  callBack,
  onError,
}: getCurrentLiveLocationType) => {
  if (!navigator) throw new Error("Navigator doesn't exist ");
  if (!navigator.geolocation) throw new Error("Geolocation doesn't exist ");
  navigator.geolocation.watchPosition((position: GeolocationPosition) => {
    callBack(position);
  }, onError);
};
