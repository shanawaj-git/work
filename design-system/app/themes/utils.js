import { getTheme } from './getTheme';

const BOX_SHADOW_MAPPING = {
  level1: '0px 3px 10px rgba(0, 0, 0, 0.07)',
  level2: '0px 6px 10px rgba(0, 0, 0, 0.07)',
  level3: '0px 10px 20px rgba(0, 0, 0,  0.07)',
  level4: '0px 14px 28px rgba(0, 0, 0, 0.1)',
  level5: '0px 19px 38px rgba(0, 0, 0, 0.15)',
  focus: `0px 0px 10px ${getTheme().theme.color.d100}`,
};

const boxShadowSelector = level => BOX_SHADOW_MAPPING[level];
const colorSelector = color => getTheme().theme.color[color.toLowerCase()];
export { boxShadowSelector, colorSelector };
