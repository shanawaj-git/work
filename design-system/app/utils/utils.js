import { v4 } from 'uuid';
import _debounce from 'lodash/debounce';
/**
 * Helper: Convert Array of coordinates to an object of cordinates
 * @param {*} cordinates : Array [lat,lng]
 * @returns
 */
export const transformedCordinates = cordinates => {
  const [lat, lng] = cordinates;
  return {
    lat,
    lng,
  };
};

export const readableCordinates = ({ lat, lng }) => ({
  lat: lat(),
  lng: lng(),
});

export const uuid = () => v4();

const MAP_ADDRESS_COMPONENT_TYPE = {
  LOCALITY: 'locality',
  COUNTRY: 'country',
};

export const getCityAndCountry = address => {
  let city;
  let country;
  // eslint-disable-next-line no-unused-expressions
  address?.forEach(component => {
    const { types } = component;
    if (types.indexOf(MAP_ADDRESS_COMPONENT_TYPE.LOCALITY) > -1) {
      city = component.long_name;
    }

    if (types.indexOf(MAP_ADDRESS_COMPONENT_TYPE.COUNTRY) > -1) {
      country = component.long_name;
    }
  });
  return {
    city,
    country,
  };
};

export const debounce = (func, wait) => _debounce(func, wait);

export const canBeRenderedInTypographyComponent = obj =>
  typeof obj === 'string' ||
  obj?.type?.displayName === 'MemoizedFormattedMessage';
