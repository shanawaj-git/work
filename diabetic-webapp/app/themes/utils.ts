
import { diabeticTheme } from './theme';

const BOX_SHADOW_MAPPING = {
  level1: '0px 3px 10px rgba(0, 0, 0, 0.07)',
  level2: '0px 6px 10px rgba(0, 0, 0, 0.07)',
  level3: '0px 10px 20px rgba(0, 0, 0,  0.07)',
  level4: '0px 14px 28px rgba(0, 0, 0, 0.1)',
  level5: '0px 19px 38px rgba(0, 0, 0, 0.15)',
};

const boxShadowSelector = (level: string): string => BOX_SHADOW_MAPPING[level];

const colorSelector = (color: string): string =>
  diabeticTheme.color[color.toLowerCase()];

export { boxShadowSelector, colorSelector };
