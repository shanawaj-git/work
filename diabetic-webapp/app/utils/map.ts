export const getAddress = (mapOutput): Object => {
  const {
    form: { direction, flatvillaname, flatvillano },
    addressFetched: {
      formatted_address,
      geometry: {
        location: { lat, lng },
      },
    },
  } = mapOutput;
  return {
    flatVillaNumber: flatvillano,
    formattedText: formatted_address,
    buildingName: flatvillaname,
    userNotes: direction,
    latitude: lat,
    longitude: lng,
  };
};
