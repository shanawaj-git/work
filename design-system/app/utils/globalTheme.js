// eslint-disable-next-line import/no-cycle
import { getTheme } from 'themes/getTheme';
import { SETTINGS } from '../config/organizations';

let currentTheme = 'DEFAULT';
const newThemes = {};

const validateTheme = type => {
  if (!SETTINGS[type])
    throw new Error('ERROR : Theme name provided is not defined');
};

class Theme {
  getInstance() {
    return this;
  }

  getTheme() {
    return currentTheme;
  }

  getDynamicThemes() {
    return newThemes;
  }

  setTheme(theme) {
    if (typeof theme === 'string') {
      validateTheme(theme);
      currentTheme = theme;
    } else {
      newThemes[theme.name] = theme;
      currentTheme = theme.name;
    }
  }

  getActiveTheme() {
    return getTheme();
  }
}
export const GlobalTheme = new Theme();
