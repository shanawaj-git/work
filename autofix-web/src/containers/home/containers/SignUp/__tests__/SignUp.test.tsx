import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { SignUp } from "../index";
import { withWrapper } from "@/utils/IntlProviderWrapper";

describe("<SignUp />", () => {
  test("it should mount", () => {
    render(withWrapper(<SignUp />));
    const signUp = screen.getByTestId("SignUp");

    expect(signUp).toBeInTheDocument();
  });
  test("it should Match the snapshot", () => {
    const { container } = render(withWrapper(<SignUp />));
    expect(container.innerHTML).toMatchSnapshot();
  });
  test("Should throw an error if input is not a correct email", () => {
    const { container, rerender, getByPlaceholderText } = render(
      withWrapper(<SignUp />)
    );
    const input = getByPlaceholderText("Your email");
    fireEvent.change(input, {
      target: {
        value: "Test Example",
      },
    });
    rerender(withWrapper(<SignUp />));
    expect(container.innerHTML).toMatchSnapshot();
  });
  test("Should not throw an error if input is a correct email", () => {
    const { container, rerender, getByPlaceholderText } = render(
      withWrapper(<SignUp />)
    );
    const input = getByPlaceholderText("Your email");
    fireEvent.change(input, {
      target: {
        value: "rajaosama.me@gmail.com",
      },
    });
    rerender(withWrapper(<SignUp />));
    expect(container.innerHTML).toMatchSnapshot();
  });
});
