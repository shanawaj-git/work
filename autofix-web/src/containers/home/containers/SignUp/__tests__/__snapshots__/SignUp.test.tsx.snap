// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> Should not throw an error if input is a correct email 1`] = `"<div data-testid=\\"SignUp\\" class=\\" text-white py-24 py-100 px-6\\"><div data-testid=\\"article-component\\"><h1 class=\\"MuiTypography-root MuiTypography-h1  !font-normal !tracking-normal css-1yzmk4e-MuiTypography-root\\" data-testid=\\"article-component-title\\"><span> Get a free booking!</span></h1><div data-testid=\\"article-component-divider\\" class=\\"w-10 h-0 border-solid border-2 my-4 border-[#E25F6D] my-4\\" style=\\"border-color: #e25f6d;\\"></div><div color=\\"white\\" class=\\" !font-normal !tracking-normal\\" typographytype=\\"largeTitle\\"><span class=\\"text-base \\">Subscribe to receive our offers and enjoy 1 booking for free! No spam, we promise.</span></div></div><div class=\\"root MuiFormControl-root styledComponents__FormControl-lwe1h6-0 jHalSF sc-edb9ce2-0 gBVPiv mt-4  css-1nrlq1o-MuiFormControl-root\\"><div class=\\"MuiFormControl-root MuiTextField-root css-1vly5ry-MuiFormControl-root-MuiTextField-root\\" data-testid=\\"email-signup-input\\"><label class=\\"MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1c2i806-MuiFormLabel-root-MuiInputLabel-root\\" data-shrink=\\"true\\" for=\\":r3:\\" id=\\":r3:-label\\"><span class=\\"text-white\\">E-mail</span></label><div class=\\"MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root\\"><input aria-invalid=\\"false\\" autocomplete=\\"\\" id=\\":r3:\\" name=\\"\\" placeholder=\\"Your email\\" type=\\"text\\" class=\\"MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input\\" value=\\"\\"></div></div></div><button class=\\"root StyledButton-sc-1apep4b-0 jVvup my-7 !normal-case MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-1fu7jd5-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" type=\\"button\\"><div style=\\"visibility: visible;\\">Sign me up</div><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button><p class=\\"MuiTypography-root MuiTypography-body1 !font-light opacity-80 text-sm css-x61q1w-MuiTypography-root\\">We wont pass your details on anyone else. By clicking sign me up button you agree to our <span style=\\"color: rgb(226, 95, 109);\\"> terms of use </span> and<span style=\\"color: rgb(226, 95, 109);\\"> privacy policy </span>.</p></div>"`;

exports[`<SignUp /> Should throw an error if input is not a correct email 1`] = `"<div data-testid=\\"SignUp\\" class=\\" text-white py-24 py-100 px-6\\"><div data-testid=\\"article-component\\"><h1 class=\\"MuiTypography-root MuiTypography-h1  !font-normal !tracking-normal css-1yzmk4e-MuiTypography-root\\" data-testid=\\"article-component-title\\"><span> Get a free booking!</span></h1><div data-testid=\\"article-component-divider\\" class=\\"w-10 h-0 border-solid border-2 my-4 border-[#E25F6D] my-4\\" style=\\"border-color: #e25f6d;\\"></div><div color=\\"white\\" class=\\" !font-normal !tracking-normal\\" typographytype=\\"largeTitle\\"><span class=\\"text-base \\">Subscribe to receive our offers and enjoy 1 booking for free! No spam, we promise.</span></div></div><div class=\\"root MuiFormControl-root styledComponents__FormControl-lwe1h6-0 jHalSF sc-edb9ce2-0 gBVPiv mt-4  css-1nrlq1o-MuiFormControl-root\\"><div class=\\"MuiFormControl-root MuiTextField-root css-1vly5ry-MuiFormControl-root-MuiTextField-root\\" data-testid=\\"email-signup-input\\"><label class=\\"MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-error MuiFormLabel-filled css-1c2i806-MuiFormLabel-root-MuiInputLabel-root\\" data-shrink=\\"true\\" for=\\":r2:\\" id=\\":r2:-label\\"><span class=\\"text-white\\">E-mail</span></label><div class=\\"MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root\\"><input aria-invalid=\\"true\\" autocomplete=\\"\\" id=\\":r2:\\" name=\\"\\" placeholder=\\"Your email\\" type=\\"text\\" class=\\"MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input\\" value=\\"\\"></div></div><p class=\\"root MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained styledComponents__FormHelperText-lwe1h6-1 jsmHnI css-1wc848c-MuiFormHelperText-root\\">Incorrect email format </p></div><button class=\\"root StyledButton-sc-1apep4b-0 jVvup my-7 !normal-case MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-1fu7jd5-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" type=\\"button\\"><div style=\\"visibility: visible;\\">Sign me up</div><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button><p class=\\"MuiTypography-root MuiTypography-body1 !font-light opacity-80 text-sm css-x61q1w-MuiTypography-root\\">We wont pass your details on anyone else. By clicking sign me up button you agree to our <span style=\\"color: rgb(226, 95, 109);\\"> terms of use </span> and<span style=\\"color: rgb(226, 95, 109);\\"> privacy policy </span>.</p></div>"`;

exports[`<SignUp /> it should Match the snapshot 1`] = `"<div data-testid=\\"SignUp\\" class=\\" text-white py-24 py-100 px-6\\"><div data-testid=\\"article-component\\"><h1 class=\\"MuiTypography-root MuiTypography-h1  !font-normal !tracking-normal css-1yzmk4e-MuiTypography-root\\" data-testid=\\"article-component-title\\"><span> Get a free booking!</span></h1><div data-testid=\\"article-component-divider\\" class=\\"w-10 h-0 border-solid border-2 my-4 border-[#E25F6D] my-4\\" style=\\"border-color: #e25f6d;\\"></div><div color=\\"white\\" class=\\" !font-normal !tracking-normal\\" typographytype=\\"largeTitle\\"><span class=\\"text-base \\">Subscribe to receive our offers and enjoy 1 booking for free! No spam, we promise.</span></div></div><div class=\\"root MuiFormControl-root styledComponents__FormControl-lwe1h6-0 jHalSF sc-edb9ce2-0 gBVPiv mt-4  css-1nrlq1o-MuiFormControl-root\\"><div class=\\"MuiFormControl-root MuiTextField-root css-1vly5ry-MuiFormControl-root-MuiTextField-root\\" data-testid=\\"email-signup-input\\"><label class=\\"MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-1c2i806-MuiFormLabel-root-MuiInputLabel-root\\" data-shrink=\\"true\\" for=\\":r1:\\" id=\\":r1:-label\\"><span class=\\"text-white\\">E-mail</span></label><div class=\\"MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root\\"><input aria-invalid=\\"false\\" autocomplete=\\"\\" id=\\":r1:\\" name=\\"\\" placeholder=\\"Your email\\" type=\\"text\\" class=\\"MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input\\" value=\\"\\"></div></div></div><button class=\\"root StyledButton-sc-1apep4b-0 jVvup my-7 !normal-case MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-1fu7jd5-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" type=\\"button\\"><div style=\\"visibility: visible;\\">Sign me up</div><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button><p class=\\"MuiTypography-root MuiTypography-body1 !font-light opacity-80 text-sm css-x61q1w-MuiTypography-root\\">We wont pass your details on anyone else. By clicking sign me up button you agree to our <span style=\\"color: rgb(226, 95, 109);\\"> terms of use </span> and<span style=\\"color: rgb(226, 95, 109);\\"> privacy policy </span>.</p></div>"`;
